inc/Engine/Cache/AdvancedCache.php

```diff
@@ -0,0 +1,171 @@
+	 *
+	 * @var WP_Filesystem_Direct
+	 */
+	private $filesystem;
+
+	/**
+	 * Instantiate the class
+	 *
+	 * @param string $template_path Absolute path to template files.
+	 */
+	public function __construct( $template_path ) {
+		$this->template_path = $template_path;
+		$this->content_dir   = rocket_get_constant( 'WP_CONTENT_DIR' );
+		$this->filesystem    = rocket_direct_filesystem();
+	}
+
+	/**
+	 * Gets the content for the advanced-cache.php file
+	 *
+	 * @since 3.6
+	 *
+	 * @return string
+	 */
+	public function get_advanced_cache_content() {
+		$content = $this->filesystem->get_contents( $this->template_path . '/advanced-cache.php' );
+		$mobile  = is_rocket_generate_caching_mobile_files() ? '$2' : '';
+		$content = preg_replace( "/('{{MOBILE_CACHE}}';)(\X*)('{{\/MOBILE_CACHE}}';)/", $mobile, $content );
+
+		$replacements = [
+			'{{WP_ROCKET_PHP_VERSION}}' => rocket_get_constant( 'WP_ROCKET_PHP_VERSION' ),
+			'{{WP_ROCKET_PATH}}'        => rocket_get_constant( 'WP_ROCKET_PATH' ),
+			'{{VENDOR_PATH}}'           => rocket_get_constant( 'WP_ROCKET_VENDORS_PATH' ),
+			'{{WP_ROCKET_CONFIG_PATH}}' => rocket_get_constant( 'WP_ROCKET_CONFIG_PATH' ),
+			'{{WP_ROCKET_CACHE_PATH}}'  => rocket_get_constant( 'WP_ROCKET_CACHE_PATH' ),
+		];
+
+		foreach ( $replacements as $key => $value ) {
+			$content = str_replace( $key, $value, $content );
+		}
+
+		/**
+		 * Filter the content of advanced-cache.php file.
+		 *
+		 * @since 2.1
+		 *
+		 * @param string $content The content that will be printed in advanced-cache.php.
+		*/
+		return (string) apply_filters( 'rocket_advanced_cache_file', $content );
+	}
+
+	/**
+	 * This warning is displayed when the advanced-cache.php file isn't writeable
+	 *
+	 * @since 3.6 Moved to a method in AdvancedCache
+	 * @since 2.0
+	 *
+	 * @return void
+	 */
+	public function notice_permissions() {
+		if ( ! $this->is_user_allowed() ) {
+			return;
+		}
+
+		if (
+			$this->filesystem->is_writable( "{$this->content_dir}/advanced-cache.php" )
+			|| rocket_get_constant( 'WP_ROCKET_ADVANCED_CACHE' )
+		) {
+			return;
+		}
+
+		$notice_name = 'rocket_warning_advanced_cache_permissions';
+
+		if (
+			in_array(
+				$notice_name,
+				(array) get_user_meta( get_current_user_id(), 'rocket_boxes', true ),
+				true
+			)
+		) {
+			return;
+		}
+
+		rocket_notice_html(
+			[
+				'status'           => 'error',
+				'dismissible'      => '',
+				'message'          => $this->get_notice_message(),
+				'dismiss_button'   => $notice_name,
+				'readonly_content' => $this->get_advanced_cache_content(),
+			]
+		);
+	}
+
+	/**
+	 * This warning is displayed when the advanced-cache.php file isn't ours
+	 *
+	 * @since 3.6 Moved to a method in AdvancedCache
+	 * @since 2.2
+	 *
+	 * @return void
+	 */
+	public function notice_content_not_ours() {
+		global $pagenow;
+
+		if (
+			'plugins.php' === $pagenow
+			&& isset( $_GET['activate'] ) // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
What about this? No more phpcs comment.
```suggestion
			&& filter_has_var( INPUT_GET, 'activate' )
```